yacl/crypto/ecc/hash_to_curve/p256.cc

```diff
@@ -0,0 +1,355 @@
+#include "yacl/crypto/ecc/hash_to_curve/p256.h"
+
+#include <array>
+#include <cstdint>
+#include <cstring>
+#include <utility>
+
+#include "yacl/base/buffer.h"
+#include "yacl/base/exception.h"
+#include "yacl/crypto/hash/ssl_hash.h"
+#include "yacl/math/mpint/mp_int.h"
+#include "yacl/utils/spi/type_traits.h"
+
+// Curve Parameters
+constexpr int kEccKeySize = 32;
+
+// p = 2^256 - 2^224 + 2^192 + 2^96 - 1
+constexpr std::array<uint8_t, kEccKeySize> p_bytes = {
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xff, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0xff, 0xff, 0xff, 0xff};
+
+constexpr std::array<uint8_t, kEccKeySize> a_bytes = {
+    0xfc, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xff, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0xff, 0xff, 0xff, 0xff};
+
+constexpr std::array<uint8_t, kEccKeySize> b_bytes = {
+    0x4b, 0x60, 0xd2, 0x27, 0x3e, 0x3c, 0xce, 0x3b, 0xf6, 0xb0, 0x53,
+    0xcc, 0xb0, 0x06, 0x1d, 0x65, 0xbc, 0x86, 0x98, 0x76, 0x55, 0xbd,
+    0xeb, 0xb3, 0xe7, 0x93, 0x3a, 0xaa, 0xd8, 0x35, 0xc6, 0x5a};
+
+constexpr std::array<uint8_t, kEccKeySize> z_bytes = {
+    0xf5, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xff, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0xff, 0xff, 0xff, 0xff};
+
+constexpr std::array<uint8_t, 32> c1_bytes = {
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x40, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x40, 0x00, 0x00, 0x00, 0xc0, 0xff, 0xff, 0xff, 0x3f};
+
+yacl::math::MPInt DeserializeMPInt(yacl::ByteContainerView buffer,
+                                   yacl::Endian endian = yacl::Endian::native) {
+  YACL_ENFORCE(buffer.size() == kEccKeySize);
+  yacl::math::MPInt mp;
+
+  mp.FromMagBytes(buffer, endian);
+
+  return mp;
+}
+
+void MPIntToBytesWithPad(unsigned char *buf, size_t buf_len,
```

Comment:
avoid using c-style pointers, and length. Use `std::string`, `Buffer`, or `ByteContianterView` instead